### PR TITLE
Added DecoderDisplay unit tests

### PR DESCRIPTION
**Pull request by hmarts9**
Added unit tests to ensure proper functionality of the DecoderDisplay code file. Tests cover everything from basic initialization to edge cases.

### Key additions:
- Basic window setup and component initialization tests
- Layout structure check for headers and data rows
- Visual property tests for labels, scroll areas, and size policies
- Data row creation and display tests (I did single and multiple rows)
- Sorting verification for TLCs
- Edge case handling for empty data sets

### Helper methods added:
- populate_rows(): Standardizes row creation across tests
- verify_layout_structure(): Makes layout verification more consistent

Test setup includes Qt application initialization/teardown and mocking of file operations.

### CODE REVIEW

File: tests/displays/cavity_display/frontend/test_decoder.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+from PyQt5.QtCore import Qt
+from displays.cavity_display.frontend.decoder import DecoderDisplay, Row, rows
+
+
+
+class TestDecoderDisplay(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Need to create a QApplication instance before running any PyQt tests
+        cls.app = QApplication([])
```
Comment (lisazacarias) [inline tests/displays/cavity_display/frontend/test_decoder.py:13]: Ok so I learned the hard way that this can cause problems if a new QApp is created in multiple test files. A quick fix is to have it inside of a `if not QApplication.instance():` check

File: tests/displays/cavity_display/frontend/test_decoder.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.app.quit()
```
Comment (lisazacarias) [inline tests/displays/cavity_display/frontend/test_decoder.py:17]: Unless this line takes care of it... I'm actually not sure

**Review by lisazacarias — APPROVED** [review]
This is great! Thanks so much for doing this!

File: tests/displays/cavity_display/frontend/test_decoder.py (hunk 2 above)
Comment (hmarts9) [reply tests/displays/cavity_display/frontend/test_decoder.py:17]: I mean it does clean up the application but there could still be conflicts if multiple test files try to create QApp simultaneously. So definitely safer to just go with your approach I agree. Plus it makes sure only one QApp is ever made.
